hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/RowCounter.java

```diff
@@ -105,9 +158,11 @@ public void map(ImmutableBytesWritable row, Result values, Context context) thro
    * @throws IOException When setting up the job fails.
    */
   public Job createSubmittableJob(Configuration conf) throws IOException {
+    conf.setBoolean(OPT_COUNT_DELETE_MARKERS, this.countDeleteMarkers);
     Job job = Job.getInstance(conf, conf.get(JOB_NAME_CONF_KEY, NAME + "_" + tableName));
     job.setJarByClass(RowCounter.class);
     Scan scan = new Scan();
+    scan.setRaw(this.countDeleteMarkers);
```

Comment:
this should not be done unless we need it. it may make jobs slower for existing users who don't even want to count delete markers. 

also how do you handle multiple version of same data now? We double count same rows? I am not sure of its consistency with current behaviour.

Reply:
>also how do you handle multiple version of same data now? We double count same rows? I am not sure of its consistency with current behaviour.
>have you tested such scenarios with this change? Please provide details and add UTs for all such cases.

I am not sure if I got you. Can you share an example of a scenario that you have in mind.

Reply:
> @NihalJain , countDeleteMarkers is boolean flag. Raw scan will be set to true only when users want delete marker counts. Otherwise it will be false.

Ah got it!

> I am not sure if I got you. Can you share an example of a scenario that you have in mind.

For example, cases like we write same row with different values 3 times for a table which supports more than 1 version

Reply:
Added tests with versioned columns as well.